### ISSUE TITLE
`min-type-alias-impl-trait` should not accept impl trait in assoc. type binding

### ISSUE THREAD

**Issue by nikomatsakis** [rust-lang/rust#86731]
The following example represents an illegal use for `min_type_alias_impl_trait`

[playground of more complete code](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=bdc2c62cd370e09e559253de5c81cf2e)

```rust
pub type FooImpl = impl Foo;
pub type BarImpl = impl Bar<Foo = FooImpl>;
```

But we are not reporting it that way.

**Comment by oli-obk** [comment]
It seems fine to me under the full type-alias-impl-trait feature as implemented. I think we should just close this issue and keep allowing the example

**Comment by nikomatsakis** [comment]
This works "as expected" now and we have tests, @oli-obk claims :)
